components/treelist/columns/menu.md

```diff
@@ -0,0 +1,316 @@
+        public int? ParentId { get; set; }
+        public string Name { get; set; }
+        public DateTime HireDate { get; set; }
+        public int Salary { get; set; }
+    }
+}
+````
+
+## Features
+
+To control the features of the Column Menu, use the `<TreeListColumnMenuSettings>` tag, nested inside the `<TreeListSettings>` tag. By default, all Column Menu features are enabled.
+
+### Column Chooser
+
+The Column Chooser in the Column Menu allows you to toggle the visibility of TreeLis columns. By default, all columns are visible under the **Columns** section of the Column Menu. To expand the menu, click the **Columns** item.
+
+The **Apply** button sets the column visibility according to the current checkbox values and closes the column menu. The **Reset** button reverts the checkbox values to their state when the column menu was opened. At this point, the user can start over, click **Apply**, or click outside the column menu to close it.
+
+* To disable the column chooser, set the `ShowColumnChooser` parameter of the `<TreeLisColumnMenuSettings>` to `false`.
+* To hide a column from the Column Chooser, set the `VisibleInColumnChooser` property of the column to `false`.
+
+### Filtering
+
+To control whether filtering is possible from the Column Menu, set the `FilterMode` parameter of the `TreeListColumnMenuSettings` tag to a member of the `ColumnMenuFilterMode` enum:
+
+* `None`—disables the filtering from the Column Menu. This is the recommended option if you use the [`FilterRow` mode]({%slug treelist-filter-row%}).
+* `FilterMenu`—enables filtering from a filter menu.
+
+### Frozen Columns
+
+To disable the locking and unlocking of a column from the Column Menu, set the `Lockable` parameter of the column to `false`.
+
+### Sorting
+
+To remove the sorting option from the Column Menu, set the `Sortable` parameter of the `TreeListColumnMenuSettings` tag to `false`.
+
+### Reorderable
+
+To allow column reordering from the Column Menu, set the `Reorderable` parameter of the `TreeListColumnMenuSettings` tag to `true`.
+
+### Sections
+
+The TreeList Column Menu lets you group the columns in the [Column Chooser](#column-chooser) into different sections:
+
+1. Use the `TreeListColumnMenuChooser` tag (child to the `TreeListColumnMenuSettings`).
+
+1. Add the [Template]({%slug treelist-templates-column-chooser%}) tag.
+
+1. Provide a `TreeListColumnMenuChooserGroup` which is a collection of the columns that will be in the section. To render a title for the section, use the `Title` parameter.
+
+1. Use the `TreeListColumnMenuChooserItem` to denote the columns that belong to the group.
+
+    * You must set the `ColumnId` parameter of the `TreeListColumnMenuChooserItem` to the value of the `Id` parameter of the corresponding TreeList Column.
+    
+    * If you set the `Title` parameter of the `TreeListColumnMenuChooserItem`, it will override the value of the `Title` parameter of the corresponding TreeList Column. 
+
+## Examples 
```

Comment:
The two child headings doesn't make it clear what is the difference between the two examples. What is the difference between "features" and "configuration"? Better use a single example with everything, and there will be no need to split the features to "basic" and "advanced".

(I see that it's like that in the Grid, which doesn't guarantee it's the best option.)